solr/core/src/java/org/apache/solr/handler/component/RealTimeGetComponent.java

```diff
@@ -609,9 +618,11 @@ public static SolrInputDocument getInputDocument(SolrCore core, BytesRef idBytes
    * @param resolveFullDocument In case the document is fetched from the tlog, it could only be a partial document if the last update
    *                  was an in-place update. In that case, should this partial document be resolved to a full document (by following
    *                  back prevPointer/prevVersion)?
+   * @param resolveRootDoc Check whether the document is part of a nested hierarchy. If so, return the whole hierarchy.
+   * @param resolveChildren Check whether the document has child documents. If so, return the document including its children.
    */
   public static SolrInputDocument getInputDocument(SolrCore core, BytesRef idBytes, AtomicLong versionReturned, boolean avoidRetrievingStoredFields,
-      Set<String> onlyTheseNonStoredDVs, boolean resolveFullDocument) throws IOException {
+      Set<String> onlyTheseNonStoredDVs, boolean resolveFullDocument, boolean resolveRootDoc, boolean resolveChildren) throws IOException {
```

Comment:
resolveRootDoc -> resolves the whole hierarchy from the root doc regardless of whether the docId provided was the root document.
resolveChildren -> resolves the specified document's child documents, regardless of whether the document was is a root document or not.